include/sys/mem_blocks.h

```diff
@@ -218,6 +218,27 @@ struct sys_multi_mem_blocks {
 int sys_mem_blocks_alloc(sys_mem_blocks_t *mem_block, size_t count,
 			 void **out_blocks);
 
+
+/**
+ * @brief Force allocation of a specified blocks in a memory block object
+ *
+ * Allocate a specified blocks in a memory block object.
+ * Note: use caution when mixing sys_mem_blocks_get and sys_mem_blocks_alloc,
+ * allocation may take any of the free memory space
```

Comment:
Danger is here: once a standard alloc is executed, it may alloc a block any address in the mem_block. Even worse, the address may vary from run to run. So when there's a need to get a memory at a specific address it may be available or not in more/less random conditions. So best practice is: don't mix get and allocs OR do all gets before first alloc.

Reply:
If that's the case, it's better to put it in the code documentation, possibly in both functions as a note or warning.